### PR TITLE
Refactor AssertJ assertions into more idiomatic ones

### PR DESCRIPTION
**Pull request by krzyk**
Converted AssertJ assetions into more redable/idiomatic (e.g. instead of `assertThat(collection.size()).equals(5)` -> `assertThat(collection).hasSize(5)`) - besides shorter and more readable code it also produces more appropriate error messages.

### CODE REVIEW

**Comment by sbrannen** [comment]
Hi @krzyk,

Thanks for the PR!

This is a task that I was planning on putting "up for grabs" via our [`status: ideal-for-contribution`](https://github.com/spring-projects/spring-framework/labels/status%3A%20ideal-for-contribution) label. So it's nice to see that you've taken the initiative on your own.

Out of curiosity, how did you perform these refactorings?

Do you have a tool that assists you, or did you create your own tool (or regular expressions) to achieve the refactorings?

For future reference, if you split up changes like this into multiple commits (either by module or perhaps preferably by type of refactoring) that will make it easier for core maintainers to process the PR.

Cheers,

Sam

**Comment by krzyk** [comment]
@sbrannen Most of those were suggested by IntelliJ IDEA plugin https://plugins.jetbrains.com/plugin/12195-concise-assertj-optimizing-nitpicker-cajon- . It is not perfect (some suggestions do not compile, some fail during runtime, AFAIR it is related to generics), but it was a real time saver.

Others I found during checking the tests manually and debugging issues from the plugin.

**Comment by philwebb** [comment]
Interesting tool, thanks for /cc'ing me @sbrannen.

**Review by simonbasle — CHANGES_REQUESTED** [review]
Thanks for the PR @krzyk! 
This is quite a large one so I have reviewed up to and including `spring-context/` root so far.

Nothing major to report. I have a few nitpicks on a small number of changes that could be even more idiomatic (for instance spots that use `.isEqualTo(x)` on array length or collection sizes where `hasSize(x)` would arguably be better)...

I haven't commented on those because GitHub is starting to have trouble loading all the changes and comments in such a large PR. We will decided if it is even worth fixing, which can be done on my side with a polish commit added to this PR.

I have added two comments that are worth your attention plus the fact that this PR has conflicts with the base branch. You can fix these now or hold off until we've reviewed the full set of changes.

(to be continued)

**Comment by simonbasle** [comment]
> You can fix these now or hold off until we've reviewed the full set of changes.

if you change anything @krzyk please do so with incremental commits and not a rebase / force-push 🙇

File: spring-aop/src/test/java/org/springframework/aop/framework/ProxyFactoryTests.java
Comment (krzyk) [reply spring-aop/src/test/java/org/springframework/aop/framework/ProxyFactoryTests.java:55]: @simonbasle Yes I was wondering about that, where is the line between cosmetic and substantial changes - so I used arbitrary line of lines of code changed with 30 lines being the line.

So should I remove the author tags? Or use some other metric for applying it?

**Comment by krzyk** [comment]
> `.isEqualTo(x)` on array length or collection sizes where `hasSize(x)`

Just a note that in some places that is not possible even if it looks like it should be - of course there are probably places which I missed.

**Comment by krzyk** [comment]
> if you change anything @krzyk please do so with incremental commits and not a rebase / force-push bow

Sure, at first I wasn't sure what's your take on the amount of commits in a single PR, some project prefer to have a single one, others like to squash them in a single one.

So I'm holding off on rebasing until you finish reviewing, I think this will be easier.

Again sorry for such big PR - it won't happen again :)

**Comment by krzyk** [comment]
@simonbasle I wonder (if you looked already at spring-context) maybe it would be better if I revert my changes to the other subprojects (in this PR) and create few other PRs with changes in those?

File: spring-aop/src/test/java/org/springframework/aop/framework/ProxyFactoryTests.java
Comment (simonbasle) [reply spring-aop/src/test/java/org/springframework/aop/framework/ProxyFactoryTests.java:55]: it's a bit of a gray area, but I don't think for (partially automated) refactors of tests it is that common. I'd go the route of removing the tags, if you don't mind.

**Comment by simonbasle** [comment]
> @simonbasle I wonder (if you looked already at spring-context) maybe it would be better if I revert my changes to the other subprojects (in this PR) and create few other PRs with changes in those?

could be interesting yeah, if you manage to split the remainder into PRs of comparable sizes. I think the most changes are in the web module, which would definitely make sense as its own PR, for a start

> Sure, at first I wasn't sure what's your take on the amount of commits in a single PR, some project prefer to have a single one, others like to squash them in a single one.

That's personal preference but I like the commits to be incremental during the review phase (so I don't loose track of what I've reviewed initially and what comes from review process). then at the end I'll squash and merge, to have a single meaningful commit with a detailed commit message.

> Again sorry for such big PR - it won't happen again :)

No worries :)

File: spring-test/src/test/java/org/springframework/test/context/junit4/concurrency/SpringJUnit4ConcurrencyTests.java
Comment (simonbasle) [inline spring-test/src/test/java/org/springframework/test/context/junit4/concurrency/SpringJUnit4ConcurrencyTests.java:2]: no other change in that file, please remove the date change

**Review by simonbasle — CHANGES_REQUESTED** [review]
I have now reviewed up to `spring-web` (not included). the general comments from the previous review still apply, but I'm less nitpicky for minor things like length.isEqualsTo vs hasSize (these are still improvements over the existing assertions anyway)

**Comment by simonbasle** [comment]
@krzyk I think I'll be able to review everything today or tomorrow, so no need to split the PR for now (but something to be kept in mind for later ones 😉)

File: spring-webmvc/src/test/java/org/springframework/web/servlet/tags/MessageTagTests.java
Comment (simonbasle) [inline spring-webmvc/src/test/java/org/springframework/web/servlet/tags/MessageTagTests.java:392]: I think that `String[]` single value in the `List` was on purpose, so this changes the semantic of the test. Could be replaced by `Collections.singletonList(new String[]...)`

**Review by simonbasle — CHANGES_REQUESTED** [review]
finally got to review everything

File: spring-webmvc/src/test/java/org/springframework/web/servlet/FlashMapTests.java
Comment (krzyk) [reply spring-webmvc/src/test/java/org/springframework/web/servlet/FlashMapTests.java]: done

File: spring-test/src/test/java/org/springframework/test/context/junit4/concurrency/SpringJUnit4ConcurrencyTests.java
Comment (krzyk) [reply spring-test/src/test/java/org/springframework/test/context/junit4/concurrency/SpringJUnit4ConcurrencyTests.java]: done

File: spring-aop/src/test/java/org/springframework/aop/framework/ProxyFactoryTests.java
Comment (krzyk) [reply spring-aop/src/test/java/org/springframework/aop/framework/ProxyFactoryTests.java]: done

File: spring-webmvc/src/test/java/org/springframework/web/servlet/tags/MessageTagTests.java
Comment (krzyk) [reply spring-webmvc/src/test/java/org/springframework/web/servlet/tags/MessageTagTests.java:392]: But `new String[]` will either way be used here. Because Arrays.asList("arg1", "arg2") is the same as Arrays.asList(new String[]{"arg1", "arg2"}). The former will be converted to the later.

It will be different with the singletonList, as it will create a list of array, while the original was creating list of `arg1` and `arg2`.

**Comment by krzyk** [comment]
@simonbasle I updated the code and merged upstream changes.

**Review by simonbasle — APPROVED** [review]
thanks for the updates, this looks good now 👍

**Pull request merged**
